fs/march/march.go

```diff
@@ -416,17 +417,29 @@ func (m *March) processJob(job listDirJob) ([]listDirJob, error) {
 	}
 
 	// If NoTraverse is set, then try to find a matching object
-	// for each item in the srcList
+	// for each item in the srcList to head dst object
+	ci := fs.GetConfig(m.Ctx)
+	limiter := make(chan struct{}, ci.Checkers)
 	if m.NoTraverse && !m.NoCheckDest {
 		for _, src := range srcList {
-			if srcObj, ok := src.(fs.Object); ok {
-				leaf := path.Base(srcObj.Remote())
-				dstObj, err := m.Fdst.NewObject(m.Ctx, path.Join(job.dstRemote, leaf))
-				if err == nil {
-					dstList = append(dstList, dstObj)
+			wg.Add(1)
+			limiter <- struct{}{}
+			go func(limiter chan struct{}, src fs.DirEntry) {
+				defer wg.Done()
+				if srcObj, ok := src.(fs.Object); ok {
+					leaf := path.Base(srcObj.Remote())
+					dstObj, err := m.Fdst.NewObject(m.Ctx, path.Join(job.dstRemote, leaf))
+					if err == nil {
+						mutex.Lock()
+						dstList = append(dstList, dstObj)
+						mutex.Unlock()
+					}
 				}
-			}
+				<-limiter
+			}(limiter, src)
 		}
+		close(limiter)
```

Comment:
You don't want to close the limiter - just remove this line.

Reply:
done